### PR TITLE
Fixed #26447 -- Deprecated settings.USE_ETAGS in favor of ConditionalGetMiddleware.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/26447

Updated from https://github.com/django/django/pull/6393

### CODE REVIEW

**Comment by timgraham** [comment]
@marfire, is this something you're interested in reviewing?

**Comment by marfire** [comment]
@timgraham: I've glanced over the code and it all looks good to me; however, I'm out of town traveling at the moment so I'm not able to give it a thorough review. If the PR is still lingering when I'm back in three weeks I'll review it properly and mark it for check in.

File: docs/ref/settings.txt [hunk 1]
```diff
@@ -2532,6 +2532,11 @@ bandwidth but slows down performance. This is used by the
 :class:`~django.middleware.common.CommonMiddleware` and in the :doc:`cache
 framework </topics/cache>`.
 
+.. deprecated:: 1.11
+
+    This setting is deprecated in favor of the ``ConditionalGetMiddleware``,
+    which sets an ETag regardless of this setting.
```
Comment (marfire) [inline docs/ref/settings.txt]: This is a minor point, but in other places the phrase seems to be just "`ConditionalGetMiddleware`" or "the `ConditionalGetMiddleware` middleware". "The `ConditionalGetMiddleware`" reads a bit strange to me. Maybe "...in favor of using `ConditionalGetMiddleware`..."

File: docs/topics/conditional-view-processing.txt [hunk 2]
```diff
@@ -189,17 +189,14 @@ every time.
...
-:class:`~django.middleware.common.CommonMiddleware`. While certainly being
-easy to use and suitable for many situations, those pieces of middleware
-functionality have limitations for advanced usage:
-
-* They are applied globally to all views in your project
-* They don't save you from generating the response itself, which may be
-  expensive
-* They are only appropriate for HTTP ``GET`` requests.
+Django provides simple and straightforward conditional ``GET`` handling via the
+:class:`django.middleware.http.ConditionalGetMiddleware`. While being easy to
```
Comment (marfire) [inline docs/topics/conditional-view-processing.txt]: As above.

**Pull request merged**